test/e2e/storage/testsuites/snapshotsecrets.go

```diff
@@ -0,0 +1,253 @@
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+type snapshotsecretsTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &snapshotsecretsTestSuite{}
+
+// InitSnapshotsecretsTestSuite returns snapshottableTestSuite that implements TestSuite interface
+func InitSnapshotsecretsTestSuite() TestSuite {
+	return &snapshotsecretsTestSuite{
+		tsInfo: TestSuiteInfo{
+			Name: "snapshotsecrets",
+			TestPatterns: []testpatterns.TestPattern{
+				// testpatterns.DynamicSnapshotDelete,
+				testpatterns.DynamicSnapshotRetain,
+				// testpatterns.PreprovisionedSnapshotDelete,
+				// testpatterns.PreprovisionedSnapshotRetain,
+			},
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: "1Mi",
+			},
+			FeatureTag: "[Feature:VolumeSnapshotDataSource]",
+		},
+	}
+}
+
+func (s *snapshotsecretsTestSuite) GetTestSuiteInfo() TestSuiteInfo {
+	return s.tsInfo
+}
+
+func (s *snapshotsecretsTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {
+}
+
+func (s *snapshotsecretsTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	ginkgo.BeforeEach(func() {
+		// Check preconditions.
+		dInfo := driver.GetDriverInfo()
+		ok := false
+		sDriver, ok = driver.(SnapshottableTestDriver)
+		if !dInfo.Capabilities[CapSnapshotDataSource] || !ok {
+			e2eskipper.Skipf("Driver %q does not support snapshots - skipping", dInfo.Name)
+		}
+		dDriver, ok = driver.(DynamicPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %q does not support dynamic provisioning - skipping", driver.GetDriverInfo().Name)
+		}
+	})
+
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("snapshottingwithsecrets")
+
+	ginkgo.Describe("volume snapshot controller", func() {
+		var (
+			err    error
+			config *PerTestConfig
+			driverCleanup func()
+			cleanupSteps []func()
+
+			cs clientset.Interface
+			dc  dynamic.Interface
+			pvc *v1.PersistentVolumeClaim
+			// sc                  *storagev1.StorageClass
+			// claimSize           string
+			// originalMntTestData string
+		)
+		init := func() {
+			cleanupSteps = make([]func(), 0)
+			// init snap class, create a source PV, PVC, Pod
+			cs = f.ClientSet
+			dc = f.DynamicClient
+
+			// Now do the more expensive test initialization.
+			config, _ = driver.PrepareTest(f)
+			cleanupSteps = append(cleanupSteps, driverCleanup)
+
+			var volumeResource *VolumeResource
+			cleanupSteps = append(cleanupSteps, func() {
+				framework.ExpectNoError(volumeResource.CleanupResource())
+			})
+
+			ginkgo.By("Create volume resource using secrets")
```

Comment:
Is the volume created with secrets?  If not, remove "using secrets" as it is confusing.